src/cmd/build.js

```diff
@@ -40,41 +40,34 @@ export type PackageCreatorParams = {
 export type PackageCreatorFn =
     (params: PackageCreatorParams) => Promise<ExtensionBuildResult>;
 
-function defaultPackageCreator(
+async function defaultPackageCreator(
   {manifestData, sourceDir, fileFilter, artifactsDir}: PackageCreatorParams
 ): Promise<ExtensionBuildResult> {
+  let id;
 
-  return new Promise(
-    (resolve) => {
-      if (manifestData) {
-        const id = getManifestId(manifestData);
-        log.debug(`Using manifest id=${id || '[not specified]'}`);
-        resolve(manifestData);
-      } else {
-        resolve(getValidatedManifest(sourceDir));
-      }
-    })
-    .then((manifestData) => {
-      return zipDir(
-        sourceDir, {
-          filter: (...args) => fileFilter.wantFile(...args),
-        })
-        .then((buffer) => {
-          let packageName = safeFileName(
-            `${manifestData.name}-${manifestData.version}.zip`);
-          let extensionPath = path.join(artifactsDir, packageName);
-          let stream = createWriteStream(extensionPath);
-          let promisedStream = streamToPromise(stream);
-
-          stream.write(buffer, () => stream.end());
-
-          return promisedStream
-            .then(() => {
-              log.info(`Your web extension is ready: ${extensionPath}`);
-              return {extensionPath};
-            });
-        });
-    });
+  if (manifestData) {
+    id = getManifestId(manifestData);
+    log.debug(`Using manifest id=${id || '[not specified]'}`);
+  } else {
+    manifestData = await getValidatedManifest(sourceDir);
+  }
+
+  let buffer =  await zipDir(sourceDir, {
+    filter: (...args) => fileFilter.wantFile(...args),
+  });
+
+  let packageName = safeFileName(
```

Comment:
FWIW, there are 2 new ESLint rules you can add:

- [`no-var`](eslint.org/docs/rules/no-var) &mdash; require `let` or `const` instead of `var`.
- [`prefer-const`](eslint.org/docs/rules/prefer-const) &mdash; Suggest using `const`.

```diff
diff --git a/.eslintrc b/.eslintrc
index 5589499..5264ac3 100644
--- a/.eslintrc
+++ b/.eslintrc
@@ -73,8 +73,10 @@
     "no-undef": 2,
     "no-underscore-dangle": 0,
     "no-unused-vars": 2,
+    "no-var": 2,
     "no-with": 2,
     "one-var": [2, "never"],
+    "prefer-const": 2,
     "prefer-template": 2,
     "quote-props": [1, "consistent-as-needed"],
     "quotes": [2, "single", "avoid-escape"],
```

After adding those 2 rules, I get `(125 errors, 0 warnings)` after running `npm run lint`.